### PR TITLE
Cleanup PARL_IO

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
This improves docs of the PARL_IO module.

I think this doesn't deserve a CHANGELOG entry. It makes a few things (especially DMA channels in some structs) more private which technically is a note-worthy change but any code which was accessing that should never have been written 😄 

#### Testing
Everything works as before - the examples compile fine

### CODE REVIEW

File: esp-hal/src/parl_io.rs [hunk 1]
```diff
@@ -1579,6 +1603,51 @@ where
     }
 }
 
+/// Creates a TX channel
+#[non_exhaustive]
```
Comment (MabezDev) [inline esp-hal/src/parl_io.rs:1607]: I don't think these need to be non_exhaustive because they don't have public fields?

**Review by MabezDev — CHANGES_REQUESTED** [review]

File: esp-hal/src/parl_io.rs (hunk 1 above)
Comment (bjoernQ) [reply esp-hal/src/parl_io.rs:1607]: true - I think I made them first non-exhaustive and then made the fields private or something like that 👍

**Review by MabezDev — APPROVED** [review]
Thank you! Loving the doc updates :star_struck:

**Review by SergioGasquez — APPROVED** [review]
LGTM! Thanks <3

**Pull request merged**